improver/calibration/load_and_train_quantile_regression_random_forest.py

```diff
@@ -46,11 +46,11 @@ def __init__(
         self,
         feature_config: dict[str, list[str]],
         parquet_diagnostic_names: Union[list[str], str],
-        target_cf_name: str,
+        cf_names: Union[list[str], str],
         forecast_periods: str,
         cycletime: str,
         training_length: int,
-        experiment: Optional[str] = None,
+        experiments: Optional[list[str]] = None,
```

Comment:
This is Optional, but if it is not set the filter below ends up as:
```
("experiment", "in", [])
```
Does that not mean that if this is not set no data can ever be read? Or does the filter do something different when provided with an empty list?

Assuming it behaves as I expect, it seems this should not be optional as without it this code will simply not function; alternatively the inclusion of `experiment` in the filters should be contingent on this value being provided in which case it can remain optional.

I imagine this was true when it was a single value or None as well (`"diagnostic", "==", None`), which I missed.